internal/ebpf/probesfacade/captor/capture.go

```diff
@@ -107,51 +119,110 @@ func Capture(functionSymbol string, cmdArgs []string, opts CaptureOptions) ([]ui
 	// init perf buffer
 	eventsChannel := make(chan []byte)
 	lostChannel := make(chan uint64)
-	rb, err := bpfModule.InitPerfBuf(bpfEventsMap, eventsChannel, lostChannel, 1)
+	pb, err := bpfModule.InitPerfBuf(bpfEventsMap, eventsChannel, lostChannel, 1)
 	if err != nil {
 		return nil, fmt.Errorf("error initializing map (%s) with PerfBuffer: %v", bpfEventsMap, err)
 	}
 
-	// run args that we want to trace
+	return &ebpfSetup{
+		mod:      bpfModule,
+		link:     traceLink,
+		pb:       pb,
+		eventsCh: eventsChannel,
+		lostCh:   lostChannel,
+		opts:     opts,
+		cmd:      cmdArgs,
+	}, nil
+}
+
+// Close closes the ebpf link and module.
+func (ebpf *ebpfSetup) Close() {
+	ebpf.link.Destroy()
+	ebpf.mod.Close()
+}
+
+// Capture collects syscalls from the executed command.
+// Returns values through the given channels.
+func (ebpf *ebpfSetup) Capture(collectedSyscalls chan []uint32, errorCh chan error) {
```

Comment:
For now this errorCh is never used, so either we use it (see comments below) or consider removing it

Reply:
I would keep it for the future, I don't want to change the signature of the function more often.